libraries/chain/include/eosio/chain/backing_store/db_key_value_iter_cache.hpp

```diff
@@ -0,0 +1,156 @@
+#pragma once
+#include <fc/utility.hpp>
+#include <sstream>
+#include <algorithm>
+#include <set>
+
+namespace eosio { namespace chain { namespace backing_store {
+struct unique_table {
+   name contract;
+   name scope;
+   name table;
+};
+
+template<typename T>
+struct secondary_key {
+   int       table_ei;
+   T         secondary;
+   uint64_t  primary;
+   uint64_t  payer;
+};
+
+struct unique_table_hash {
+   std::size_t operator()(const unique_table& t) const {
+      return std::hash<name>()(t.contract) ^ std::hash<name>()(t.scope) ^ std::hash<name>()(t.table);
```

Comment:
`contract == scope` is fairly common and will cause them to cancel.

Reply:
Please verify you are ok with secondary_key operator<

Reply:
operator< can be implemented using std::tie, which is easier to understand.